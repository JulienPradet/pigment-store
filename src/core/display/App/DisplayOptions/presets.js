export default [{
  label: 'S',
  preset: {
    size: {
      width: '360',
      height: 'auto'
    },
    zoom: '100'
  }
}, {
  label: 'M',
  preset: {
    size: {
      width: '960',
      height: 'auto'
    },
    zoom: '100'
  }
}, {
  label: 'L',
  preset: {
    size: {
      width: '1920',
      height: 'auto'
    },
    zoom: '50'
  }
}]
