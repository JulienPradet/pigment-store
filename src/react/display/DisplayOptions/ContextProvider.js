import React from 'react'
import {compose, withState, withContext, getContext} from 'recompose'

const displayOptionsContextType = {
  displayOptions: React.PropTypes.object.isRequired,
  setDisplayOptions: React.PropTypes.func.isRequired
}

export const DisplayOptionsProvider = compose(
  withState('displayOptions', 'setDisplayOptions', {
    size: {
      width: '1000',
      height: 'auto'
    },
    zoom: '50'
  }),
  withContext(
    displayOptionsContextType,
    ({displayOptions, setDisplayOptions}) => ({
      displayOptions,
      setDisplayOptions
    })
  )
)(({children}) => React.Children.only(children))

export const getDisplayOptions = () => getContext(displayOptionsContextType)
