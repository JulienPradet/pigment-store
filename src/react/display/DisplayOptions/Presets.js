import React from 'react'
import {Group, Item} from '../util/View/HorizontalMenu'

const randomPreset = () => {
  const width = parseInt(Math.random() * 2000) + 300

  const zoom = window.innerWidth * 0.8 < width ? (window.innerWidth * 0.8 / width * 100).toFixed(0) : '100'

  return ({
    size: {
      width: width,
      height: 'auto'
    },
    zoom: zoom
  })
}

export default ({presets, onSelect}) => <Group>
  {presets.map(({name, preset}) => <Item key={name}>
    <button onClick={() => onSelect(preset)}>{name}</button>
  </Item>)}
  <Item>
    <button onClick={() => onSelect(randomPreset())}>Random</button>
  </Item>
</Group>
