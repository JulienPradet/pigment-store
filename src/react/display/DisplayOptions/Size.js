import React from 'react'
import {Group, Item} from '../util/View/HorizontalMenu'

const Size = ({size, onChange}) => <Group>
  <Item>
    <input type='text' value={size.width} onChange={(event) => onChange({ width: event.currentTarget.value })} />
    px
  </Item>
  <Item>
    <input type='text' value={size.height} onChange={(event) => onChange({ height: event.currentTarget.value })} />
    px
  </Item>
</Group>

export default Size
