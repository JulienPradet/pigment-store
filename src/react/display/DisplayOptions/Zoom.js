import React from 'react'
import {Group, Item} from '../util/View/HorizontalMenu'

const Zoom = ({zoom, onChange}) => <Group>
  <Item>
    <input type='number' step='5' value={zoom} onChange={(event) => onChange({ zoom: event.currentTarget.value })} />
    %
  </Item>
</Group>

export default Zoom
