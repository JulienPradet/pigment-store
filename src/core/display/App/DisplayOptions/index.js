import React from 'react'
import {compose, withHandlers} from 'recompose'
import Size from '../util/View/HorizontalMenu/Size'
import Zoom from '../util/View/HorizontalMenu/Zoom'
import PresetList from '../util/View/HorizontalMenu/Preset'
import {getDisplayOptions} from './ContextProvider'
import {Container, Group} from '../util/View/HorizontalMenu'
import presets from './presets'

const DisplayOptions = ({displayOptions, updateOptions, setOptions}) => <Container>
  <Group>
    <Size size={displayOptions.size} onChange={updateOptions} />
  </Group>
  <Group>
    <Zoom zoom={displayOptions.zoom} onChange={updateOptions} />
  </Group>
  <Group>
    <PresetList selected={displayOptions} presets={presets} onSelect={setOptions} />
  </Group>
</Container>

const SmartDisplayOptions = compose(
  getDisplayOptions(),
  withHandlers({
    updateOptions: ({setDisplayOptions, displayOptions}) => (options) => {
      setDisplayOptions(Object.assign(
        {},
        displayOptions,
        options
      ))
    },
    setOptions: ({setDisplayOptions}) => (options) => {
      setDisplayOptions(options)
    }
  })
)(DisplayOptions)

SmartDisplayOptions.displayName = 'DisplayOptions'

export default SmartDisplayOptions
