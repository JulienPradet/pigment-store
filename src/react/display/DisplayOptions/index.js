import React from 'react'
import {compose, withHandlers} from 'recompose'
import Size from './Size'
import Zoom from './Zoom'
import {PresetList} from '../util/View/HorizontalMenu/Preset'
import {getDisplayOptions} from './ContextProvider'
import {Container} from '../util/View/HorizontalMenu'
import presets from './presets'

const DisplayOptions = ({displayOptions, updateOptions, setOptions}) => <Container>
  <Size size={displayOptions.size} onChange={updateOptions} />
  <Zoom zoom={displayOptions.zoom} onChange={updateOptions} />
  <PresetList selected={displayOptions} presets={presets} onSelect={setOptions} />
</Container>

export default compose(
  getDisplayOptions(),
  withHandlers({
    updateOptions: ({setDisplayOptions, displayOptions}) => (options) => {
      setDisplayOptions(Object.assign(
        {},
        displayOptions,
        options
      ))
    },
    setOptions: ({setDisplayOptions}) => (options) => {
      setDisplayOptions(options)
    }
  })
)(DisplayOptions)
