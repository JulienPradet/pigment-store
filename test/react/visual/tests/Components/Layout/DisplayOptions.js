import React from 'react'
import PigmentStore from 'pigment-store'
import displayOptionsDecorator from '../../__decorators__/.displayOptions.decorator'
import DisplayOptions from '../../../../../../src/core/display/App/DisplayOptions/index'

export default PigmentStore.React.describe('DisplayOptions', DisplayOptions)
  .setDescription(`
The left navigation that allows to explore the styleguide
  `)
  .addDecorator(displayOptionsDecorator)
  .featureJsx(
    'Default',
    (
      <DisplayOptions />
    )
  )
