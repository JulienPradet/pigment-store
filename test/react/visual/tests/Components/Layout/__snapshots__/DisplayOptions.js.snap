exports[`DisplayOptions Default 1`] = `
<div
  className="container">
  <div
    className="group">
    <div
      className="size">
      <div
        className="sizeInput">
        <div>
          <span
            checked={false}
            className="rc-switch"
            defaultChecked={false}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            tabIndex="0">
            <span
              className="rc-switch-inner" />
          </span>
        </div>
        <div>
          Width
        </div>
        <div>
          <input
            disabled={true}
            onChange={[Function]}
            type="number"
            value={0} />
          px
        </div>
      </div>
      <div
        className="sizeInput">
        <div>
          <span
            checked={false}
            className="rc-switch"
            defaultChecked={false}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            tabIndex="0">
            <span
              className="rc-switch-inner" />
          </span>
        </div>
        <div>
          Height
        </div>
        <div>
          <input
            disabled={true}
            onChange={[Function]}
            type="number"
            value={0} />
          px
        </div>
      </div>
    </div>
  </div>
  <div
    className="group">
    <div
      className="zoom">
      <div
        className="zoomItem">
        Zoom
        <input
          onChange={[Function]}
          step="5"
          type="number"
          value={100} />
        %
      </div>
      <div
        className="zoomItem zoomSlider">
        <div
          className="rc-slider"
          onMouseDown={[Function]}
          onTouchStart={[Function]}>
          <div
            className="rc-slider-rail" />
          <div
            className="rc-slider-track rc-slider-track-1"
            style={
              Object {
                "left": "0%",
                "visibility": "visible",
                "width": "20%",
              }
            } />
          <div
            className="rc-slider-step" />
          <div
            className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "left": "20%",
              }
            } />
          <div
            className="rc-slider-mark" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="group">
    <div
      className="presetList">
      <div
        className="preset">
        <button
          onClick={[Function]}>
          S
        </button>
      </div>
      <div
        className="preset">
        <button
          onClick={[Function]}>
          M
        </button>
      </div>
      <div
        className="preset">
        <button
          onClick={[Function]}>
          L
        </button>
      </div>
    </div>
  </div>
</div>
`;
